src/widget/wfindonmenu.cpp

```diff
@@ -0,0 +1,218 @@
+#include "widget/wfindonmenu.h"
+
+#include <QDesktopServices>
+#include <QScreen>
+#include <QUrl>
+#include <QUrlQuery>
+#include <QtDebug>
+
+#include "track/track.h"
+#include "util/math.h"
+#include "util/qt.h"
+#include "util/widgethelper.h"
+
+namespace {
+
+inline QString actionPrefixSuffixSeparator(const QString propertyType, const QString& text) {
+    return propertyType + QStringLiteral(" | ") + text;
+}
+
+inline QString searchUrlSoundCloudArtist() {
+    return QString("https://soundcloud.com/search/people?");
+}
+
+inline QString searchUrlSoundCloudTitle() {
+    return QString("https://soundcloud.com/search/sounds?");
+}
+
+inline QString searchUrlSoundCloudAlbum() {
+    return QString("https://soundcloud.com/search/albums?");
+}
+
+inline QString searchUrlDiscogsGen() {
+    return QString("https://www.discogs.com/search/?");
```

Comment:
My Discogs use case is searching for: 
`https://www.discogs.com/search/?q=The%20Cure%20-%20Friday%20I%27m%20In%20Love&type=release`
This gives good Result in this case. 
`https://www.discogs.com/search/?type=release&title=Dancing+Queen&artist=ABBA` 
Finds only exact matches, which is probably not what we want, but OK in this case.  

I also like the "Artist" only search, to look up track from a particular artist. 

I am not sure about the "Album" only and the "Track" only search. This reveals similar named albums and tracks. 
Not too useful if the album has a common name like "Greatest Hits"   

Do we have a real use case for this?

Do we have other interesting use cases?

Reply:
We can have a specific search for discogs, which can have multiple queries. So that will give us better results.

Reply:
I think the use case question is valid for all lookups:
* For Title only: find remixes 
* For Album only: no idea, remove? 
* Artist + Track: lookup additional info.

Reply:
Should Title only search configured to find remixes? Or should it stay still?

I think, that would be nice to have just album to look up if there are any other tracks in that album. But I can remove it too.

Artist + Track lookup would be so nice, we can have exact search and not lost in similar artists and tracks. But I have checked now, only discogs have search with multiple queries. Others accepts just a query which is "q=", artist and track search is done over different pagination.

Reply:
> Should Title only search configured to find remixes? Or should it stay still?

Is this possible? Currently it reveals all titles with the same name, the original Track is at a random place in the result. 

> I think, that would be nice to have just album to look up if there are any other tracks in that album. 

For that use case you have to look up the Artist + Album. Else you the desired album might be listed among many by-catch.

> Artist + Track lookup would be so nice, we can have exact search and not lost in similar artists and tracks. 

I think when we just append Artist + Title, the result will Be good for most services. Check it out.

Reply:
Sounds good. 

The Remix addition works only sometimes. I think If we just using the title, remixes will be in the result.

Especially on SoundCloud it works perfect.